scripts/driverProviderAttachSession.js

```diff
@@ -2,105 +2,130 @@
 
 'use strict';
 
-var http = require('http'),
-    spawn = require('child_process').spawnSync;
+const http = require('http');
+const child_process = require('child_process');
 
-var sessionId = '';
-
-// 1. Create a new selenium session.
-var postData = JSON.stringify(
-  {'desiredCapabilities': {'browserName': 'firefox'}});
-var createOptions = {
-  hostname: 'localhost',
-  port: 4444,
-  path: '/wd/hub/session',
-  method: 'POST',
-  headers: {
-    'Content-Type': 'application/x-www-form-urlencoded',
-    'Content-Length': Buffer.byteLength(postData)
-  }
-};
-var req = http.request(createOptions, function(res) {
-  res.on('data', setBody);
-  res.on('end', checkSession);
-});
-req.write(postData);
-req.end();
-
-// 2. After making the request to create a selenium session, read the selenium
-// session id.
-var setBody = function(chunk) {
-  var body = chunk.toString();
-  sessionId = JSON.parse(body).sessionId;
+// Delete session method to be used at the end of the test as well as
+// when the tests fail.
+const deleteSession = async (sessionId, err) => {
+  await new Promise(resolve => {
```

Comment:
`await` is redundant here, because async function will return `Promise<something>` with `await` or without.

Reply:
I think this should be
```
const deleteSession = (sessionId, err) => {
  return new Promise(resolve => {
```